src/coingecko/Coingecko.ts

```diff
@@ -89,6 +89,37 @@ export class Coingecko {
     if (result.prices) return result.prices;
     throw new Error("Something went wrong fetching coingecko prices!");
   }
+
+  /**
+   * Get the current price of a token denominated in `currency`.
+   * @param contractAddress The L1 token address to fetch the price for.
+   * @param date A datestring in the format "dd-mm-yyyy" to fetch the price for.
+   * @param platform_id The platform id to fetch the price from. Defaults to "ethereum".
+   * @param currency The currency to fetch the price in. Defaults to "usd".
+   * @returns The price of the token at the given date.
+   * @throws An error if the date is invalid (malformed or in the future).
+   */
+  getContractHistoricDayPrice(contractAddress: string, date: string, platform_id = "ethereum", currency = "usd") {
+    // Resolve the date into starting/ending timestamps for the day and validate the date.
+    const [day, month, year] = date.split("-").map(Number);
+    const startDate = new Date(year, month - 1, day);
+    const endDate = new Date(year, month - 1, day, 23, 59, 59, 999); // Use 999 ms for end of the day
+    const startTimestamp = Math.floor(startDate.getTime() / 1000);
+    const endTimestamp = Math.floor(endDate.getTime() / 1000);
+    assert(isNaN(startTimestamp) || isNaN(endTimestamp) || startDate > new Date(), "Invalid date provided");
```

Comment:
I don't understand this `assert` - the logic looks inverted to me :thinking:. Why is it OK for `startTimestamp` or `endTimestamp` to be NaN? I'd also expect `startDate` to pre-date `new Date()`.

I feel like I'm missing something obvious :\

Reply:
> the logic looks inverted to me

➕

Reply:
Updated - the response of the market range didn't prove to have parity with the scraper so we should use the same API surface and move to eventually deprecate the non-contract coingecko api calls

Reply:
In the end - we did not need this check since the datetimes should error themselves out given that the endpoint will fail if we pass invalid dates